pkg/controller/node/nodecontroller.go

```diff
@@ -555,6 +565,88 @@ func (nc *NodeController) Run(stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+// addOrRemoveNoScheduleTaintByCondition taint or remove NoSchedule taint on this node based on it's condition change.
+// return:
+// 1. if node changed to schedulable
+// 2. if node changed to unschedulable
+func (nc *NodeController) addOrRemoveNoScheduleTaintByCondition(node *v1.Node) (bool, bool) {
```

Comment:
I was thinking is it really needed to return two values? Those two values can not be same, it can only be "true, false" or "false, true", why not only return one value instead?